src/OMPLPlanner.cpp

```diff
@@ -332,19 +332,49 @@ OpenRAVE::PlannerStatus OMPLPlanner::PlanPath(OpenRAVE::TrajectoryBasePtr ptraj)
 
     boost::chrono::steady_clock::time_point const tic
        = boost::chrono::steady_clock::now();
-    
+
     OpenRAVE::PlannerStatus planner_status;
+    
     try {
         // TODO: Configure anytime algorithms to keep planning.
         //m_simpleSetup->getGoal()->setMaximumPathLength(0.0);
 
-        m_simple_setup->solve(m_parameters->m_timeLimit);
-
-        if (m_simple_setup->haveExactSolutionPath()) {
+        ompl::base::PlannerStatus ompl_status;
+        ompl_status = m_simple_setup->solve(m_parameters->m_timeLimit);
+        
+        // Handle OMPL return codes, set planner_status and ptraj
+        switch ((ompl::base::PlannerStatus::StatusType)ompl_status) {
+        case ompl::base::PlannerStatus::APPROXIMATE_SOLUTION:
+            if (!m_simple_setup->haveExactSolutionPath())
+            {
+                RAVELOG_ERROR("Planner returned %s, but not path found!\n", ompl_status.asString().c_str());
+                planner_status = OpenRAVE::PS_Failed;
+                break;
+            }
+            ToORTrajectory(m_robot, m_simple_setup->getSolutionPath(), ptraj);
+            planner_status = OpenRAVE::PS_InterruptedWithSolution;
+            break;
+        case ompl::base::PlannerStatus::EXACT_SOLUTION:
+            if (!m_simple_setup->haveExactSolutionPath())
+            {
+                RAVELOG_ERROR("Planner returned %s, but not path found!\n", ompl_status.asString().c_str());
+                planner_status = OpenRAVE::PS_Failed;
+                break;
+            }
             ToORTrajectory(m_robot, m_simple_setup->getSolutionPath(), ptraj);
             planner_status = OpenRAVE::PS_HasSolution;
-        } else {
+            break;
+        default:
+            // Intended to handle:
+            // - PlannerStatus::INVALID_START
+            // - PlannerStatus::INVALID_GOAL
+            // - PlannerStatus::UNRECOGNIZED_GOAL_TYPE
+            // - PlannerStatus::CRASH
+            //  -PlannerStatus::ABORT
+            // - PlannerStatus::TIMEOUT
```

Comment:
@mkoval and I chatted about this -- this was refactored from explicit to `default` because different versions of OMPL support different error flags.